packages/emptystate/src/react/resize-observer/index.js

```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import ResizeObserverAPI from 'resize-observer-polyfill'
+
+const isBrowser = typeof window !== 'undefined'
+
+function debounce(delay, fn) {
+  let timerId
+
+  return function(...args) {
+    if (timerId) clearTimeout(timerId)
+
+    timerId = setTimeout(() => {
+      fn(...args)
+      timerId = null
+    }, delay)
+  }
+}
+
+class ResizeObserver extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.handleResize = this.handleResize.bind(this)
+    this.handleResizedObserved = debounce(
+      props.debounceDelay,
+      this.handleResizedObserved.bind(this)
+    )
+
+    this.animationId = null
+    this.container = React.createRef()
```

Comment:
this is a 16.3(i think) api...if that is going to cause problems I can take another approach

Reply:
It's hard not to want to go for the 16.x approach, since we're really close to being able to say that we fully support it and are moving off of 15.x.  

What do you think we should do on this point?

Reply:
so you create a ref in the constructor(empty at that point i guess) and then when rendering you add it as a prop to an element. you then have access to the element node by doing `this.refName.current`

Reply:
uh...I want to just release with a react 16.3 peerDependency and also have generally higher react versions going forward(especially with new components). but i also don't what to stop any team from upgrading. conflicted...